### PR TITLE
Add flyway variables support via placeholder templater

### PR DESCRIPTION
**Pull request by srjonemed**
### Brief summary of the change made

Proposed solution for #4025 

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/core/config.py
Comment (barrywhart) [inline src/sqlfluff/core/config.py:305]: Why was this added?

File: test/fixtures/templater/placeholder_flyway_var/placeholder_flyway_var_a.sql
Comment (barrywhart) [inline test/fixtures/templater/placeholder_flyway_var/placeholder_flyway_var_a.sql:1]: I don't see any mention of `flyway` in the CI output. Perhaps you need to change something else to make this run? The `placeholder` templater tests are in this file: `test/core/templaters/placeholder_test.py`.

**Review by barrywhart — CHANGES_REQUESTED** [review]

File: src/sqlfluff/core/config.py
Comment (srjonemed) [reply src/sqlfluff/core/config.py:305]: Per the configparser doc, "The first occurrence of a delimiting substring on a line is considered a delimiter" and it assumes both ':' and '=' to be delimiters by default. This would mean that the defining a key called "flyway:database = test_db_name" would split key="flyway" and val="database=test_db_name" :(
I see that all our keys use '=' as delimiter and ':' is only used in section names hence made this change. But, If there's a better way to not impose this globally, pls do let me know

File: test/fixtures/templater/placeholder_flyway_var/placeholder_flyway_var_a.sql
Comment (srjonemed) [reply test/fixtures/templater/placeholder_flyway_var/placeholder_flyway_var_a.sql:1]: thx, will take a look

Comment (srjonemed) [reply test/fixtures/templater/placeholder_flyway_var/placeholder_flyway_var_a.sql:1]: updated with a test, pls take a look

**Review by barrywhart — APPROVED** [review]
If the tests pass, this looks good to me. I'd perhaps like to take a look from my computer (currently on my phone) before merging.

**Comment by srjonemed** [comment]
Thanks!

File: test/fixtures/templater/placeholder_flyway_var/placeholder_flyway_var_a.sql
Comment (srjonemed) [reply test/fixtures/templater/placeholder_flyway_var/placeholder_flyway_var_a.sql:1]: Curious, could u let me know if this fixture is needed - not sure where/how it'd be used

**Comment by srjonemed** [comment]
@barrywhart i've pushed a [new commit](https://github.com/sqlfluff/sqlfluff/pull/4026/commits/1395d953b9f70430ade6a2c50ce6b3187bb80bdc) including adding another test. pls have a look.. 
Also, do we know when this pr could get merged and to what version?

File: src/sqlfluff/core/config.py
Comment (barrywhart) [reply src/sqlfluff/core/config.py:305]: You mentioned adding another test. Will any of your tests fail if this `delimiters` change is removed? I think it's important to have a test for this.

**Comment by barrywhart** [comment]
See my comment about the need to add a config test for the `=` delimiter. If we can address that, the PR is ready for merge.

I think 1.4.2 will be released in the next few days.

File: test/core/templaters/placeholder_test.py
Comment (barrywhart) [inline test/core/templaters/placeholder_test.py:326]: Why does `flyway_var` appear twice?

File: test/fixtures/templater/placeholder_flyway_var/placeholder_flyway_var_a.sql
Comment (barrywhart) [reply test/fixtures/templater/placeholder_flyway_var/placeholder_flyway_var_a.sql:1]: BTW, sorry it took me a while to respond. I had actually written the comment above several days ago, but I forgot to submit the pending comments. Occasionally GitHub confuses me. 😢

**Review by barrywhart — CHANGES_REQUESTED** [review]
Requesting addition of a test for the `=` delimiter

File: test/fixtures/templater/placeholder_flyway_var/placeholder_flyway_var_a.sql
Comment (srjonemed) [reply test/fixtures/templater/placeholder_flyway_var/placeholder_flyway_var_a.sql:1]: Thanks for the pointers, I'll have take a look soon

Comment (srjonemed) [reply test/fixtures/templater/placeholder_flyway_var/placeholder_flyway_var_a.sql:1]: added a test to config_test.py that should hopefully cover the delimiters codecov

File: test/core/templaters/placeholder_test.py
Comment (srjonemed) [reply test/core/templaters/placeholder_test.py:326]: I saw some warning to the effect that `ids` list has less elements than the test sql's list - let me remove and retest

**Comment by srjonemed** [comment]
> Requesting addition of a test for the `=` delimiter

added `test__config__load_placeholder_cfg` to config_test.py, pls have a look if this satisfies the gap

**Review by barrywhart — APPROVED** [review]
Looks good! I temporarily removed the `delimiters` addition and verified that your new test fails. This will prevent future maintainers from simply removing that change and thinking it's safe because no tests fail.

Unfortunately, the new release happened earlier today, so this will have to wait for the next release (usually 2-4 weeks apart).

**Comment by srjonemed** [comment]
sounds good, thx!

**Pull request merged**
